control-plane/agents/src/bin/core/volume/operations_helper.rs

```diff
@@ -527,6 +527,48 @@ impl OperationGuardArc<VolumeSpec> {
         Ok(created_replicas)
     }
 
+    pub(super) async fn resize_volume_replicas(
+        &self,
+        registry: &Registry,
+        replicas: &Vec<ResourceMutex<ReplicaSpec>>,
+        requested_size: u64,
+        spec_clone: &VolumeSpec,
+    ) -> Result<(), SvcError> {
+        for replica in replicas {
+            let mut replica = match replica.operation_guard_wait().await {
+                Ok(r) => r,
+                Err(e) => {
+                    return self
+                        .validate_update_step(registry, Err(e), spec_clone)
+                        .await
+                }
+            };
+
+            if let Some(node) = ResourceSpecsLocked::replica_node(registry, replica.as_ref()).await
+            {
+                let ret = replica
+                    .resize(
+                        registry,
+                        &ResizeReplica::new(
+                            &node,
+                            replica.as_ref().pool_name(),
+                            None,
+                            replica.uuid(),
+                            requested_size,
+                        ),
+                    )
+                    .await;
+                self.validate_update_step(registry, ret, spec_clone).await?;
+            } else {
+                let ecode = Err(SvcError::Internal {
```

Comment:
I think the scheduling should be reversed, and returned the replica states which are to be resized, and we don't need to handle this condition, it'd be handled in the scheduling which is what usually happens here.

Reply:
> I think the scheduling should be reversed, and returned the replica states which are to be resized, and we don't need to handle this condition, it'd be handled in the scheduling which is what usually happens here.

Modified, ptal. If the scheduling doesn't return all replicas as `resizable_replicas`, error will be thrown now, else we proceed to use the `resizable_replicas`

Reply:
> E               HTTP response body: {"details":"","message":"SvcError :: NoCapacityToResizeRepl: Replicas [\"4f791aeb-0f97-42fa-a43b-ceb375a88c96\"] can't be resized due to required capacity (52428810) not available on pool(s).","kind":"ResourceExhausted"}